frontend/lib/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx

```diff
@@ -144,35 +151,51 @@ export class DeckGlJsonChart extends PureComponent<PropsWithHeight, State> {
     return null
   }
 
-  static getDeckObject = (props: PropsWithHeight): DeckObject => {
-    const { element, width, height, theme } = props
-    const json = JSON.parse(element.json)
+  static getDeckObject = (
+    props: PropsWithHeight,
+    state: Partial<State>
+  ): DeckObject => {
+    const { element, width, height, theme, isFullScreen } = props
+    const { id } = state
```

Comment:
nit: I think it gets a bit more readable if we do not extract `id` here from state and instead just use `state.id` in line 165.